conda/cli/common.py

```diff
@@ -263,3 +267,51 @@ def validate_prefix(prefix):
         raise EnvironmentLocationNotFound(prefix)
 
     return prefix
+
+
+def validate_subdir_config():
+    """Validates that the configured subdir is ok. A subdir that is different from
+    the native system is only allowed if it comes from the global configration, or
+    from an environment variable.
+
+    :raises OperationNotAllowed: Active environment is not allowed to request
+                                 non-native platform packages
+    """
+    if context.subdir != context._native_subdir():
+        # We will only allow a different subdir if it's specified by global
+        # configuration, environment variable or command line argument. IOW,
+        # prevent a non-base env configured for a non-native subdir from leaking
+        # its subdir to a newer env.
+        context_sources = context.collect_all()
+        if context_sources.get("cmd_line", {}).get("subdir") == context.subdir:
+            pass  # this is ok
+        elif context_sources.get("envvars", {}).get("subdir") == context.subdir:
+            pass  # this is ok too
+        # config does not come from envvars or cmd_line, it must be a file
+        # that's ok as long as it's a base env or a global file
+        elif not paths_equal(context.active_prefix, context.root_prefix):
+            # this is only ok as long as it's base environment
```

Comment:
```suggestion
            # this is only ok as long as it's NOT base environment
```

If I'm reading correctly?